crates/scarb-api/src/lib.rs

```diff
@@ -75,64 +78,121 @@ fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
+            })
+        };
+
+    test_targets
+        .iter()
+        .filter_map(|(target_name, metadata)| starknet_artifacts_file_name(target_name, metadata))
+        .collect()
+}
+
 /// Try getting the path to `starknet_artifacts.json` file that is generated by `scarb build` or `scarb build --test` commands.
 /// If contract artifacts are produced as part of the test target and exist in both `unittest` and `integrationtest`, then the path to `integrationtest` will be returned.
 /// If the file is not present, `None` is returned.
 fn get_starknet_artifacts_path(
     target_dir: &Utf8Path,
     target_name: &str,
-    current_profile: &str,
-    use_test_target_contracts: bool,
-) -> Option<Utf8PathBuf> {
-    let starknet_artifacts_file_name = |test_type: Option<&str>| -> String {
-        if let Some(test_type) = test_type {
-            format!("{target_name}_{test_type}test.test.starknet_artifacts.json")
-        } else {
-            format!("{target_name}.starknet_artifacts.json")
-        }
-    };
-
-    let check_path = |file_name: &str| -> Option<Utf8PathBuf> {
-        let path = target_dir.join(current_profile).join(file_name);
-
-        if path.exists() {
-            Some(path)
-        } else {
-            None
-        }
-    };
-
-    if use_test_target_contracts {
-        ["integration", "unit"]
-            .iter()
-            .find_map(|test_type| check_path(&starknet_artifacts_file_name(Some(test_type))))
-    } else {
-        check_path(&starknet_artifacts_file_name(None))
-    }
+) -> Option<ContractArtifactData> {
+    let path = format!("{target_name}.starknet_artifacts.json");
+    let path = build_target_path(target_dir, &path);
+    path.map(|path| ContractArtifactData {
+        path,
+        test_type: None,
+    })
 }
 
 /// Get the map with `StarknetContractArtifacts` for the given package
 pub fn get_contracts_artifacts_and_source_sierra_paths(
     metadata: &Metadata,
-    package: &PackageId,
-    profile: Option<&str>,
+    target_dir: &Utf8Path,
+    package: &PackageMetadata,
     use_test_target_contracts: bool,
 ) -> Result<HashMap<String, (StarknetContractArtifacts, Utf8PathBuf)>> {
-    let target_name = target_name_for_package(metadata, package)?;
-    let target_dir = target_dir_for_workspace(metadata);
-    let maybe_contracts_path = get_starknet_artifacts_path(
-        &target_dir,
-        &target_name,
-        profile.unwrap_or(metadata.current_profile.as_str()),
-        use_test_target_contracts,
-    );
-
-    let map = match maybe_contracts_path {
-        Some(contracts_path) => load_contracts_artifacts_and_source_sierra_paths(&contracts_path)?,
-        None => HashMap::default(),
+    let contracts_paths = if use_test_target_contracts {
+        let test_targets = test_targets_by_name(package);
+        get_starknet_artifacts_paths_from_test_targets(target_dir, &test_targets)
+    } else {
+        let target_name = target_name_for_package(metadata, &package.id)?;
```

Comment:
This function is not needed anymore together with `compilation_unit_for_package`. Here we should instead look for `starknet-contract` target name only and return `Option`. We can also drop passing whole metadata to this function, just do
```rs
let starknet_target_name = package.targets.iter().find(|target| target.kind == "starknet-contract).map(|target| target.name);
```

 If not found, we should return no artifacts and emit a warning that `starknet-contract` target was not defined but `--no-optimization` flag was used and no contracts will be available for declare